projects/core/src/cart/facade/active-cart.service.ts

```diff
@@ -238,19 +263,12 @@ export class ActiveCartService implements OnDestroy {
     }
   }
 
-  private load(cartId: string): void {
-    if (this.userId !== OCC_USER_ID_ANONYMOUS) {
+  private load(cartId: string, userId: string): void {
+    // We want to always in every case apart of anonymous user and current cart
+    if (!(userId === OCC_USER_ID_ANONYMOUS && cartId === OCC_CART_ID_CURRENT)) {
       this.multiCartService.loadCart({
-        userId: this.userId,
-        cartId: cartId ? cartId : OCC_CART_ID_CURRENT,
-        extraData: {
-          active: true,
-        },
-      });
-    } else if (cartId && cartId !== OCC_CART_ID_CURRENT) {
-      this.multiCartService.loadCart({
-        userId: this.userId,
-        cartId: cartId,
+        userId,
+        cartId,
```

Comment:
I'm having hard time trying to understand the changes that happened here. Was your intention to preserve previous logic and just refactor it? Or some important change in the logic happened.

If we resolve above question, I have 2 nitpicks:

- I have problem understanding the comment sentence _We want to **always in every case apart of** anonymous user and current cart_
- For readability, I'd preffer resolving negations inside parenthesis (or move the condition to separate method) `userId !== OCC_USER_ID_ANONYMOUS || cartId !== OCC_CART_ID_CURRENT)`

Reply:
I started with this condition notation, but found it harder to decode that te current solution.

Reply:
Confirmed offline: nothing changed. only refactor.